prog/ai/inference_endpoint_nexus.rb

```diff
@@ -75,7 +75,8 @@ def self.assemble(project_id:, location:, boot_image:, name:, vm_size:, storage_
 
   def before_run
     when_destroy_set? do
-      unless ["destroy", "self_destroy"].include?(strand.label)
+      unless inference_endpoint.destroying_set?
+        inference_endpoint.incr_destroying
```

Comment:
Maybe we should have `incr_destroying` set at the top of the `destroy` method?  Having it set only in `before_run` will miss cases where we `hop_destroy` without `incr_destroy` being set (maybe there are no such cases currently, but it still seems safer).

Reply:
It makes sense, and it's also what Furkan suggested.
https://github.com/ubicloud/ubicloud/commit/b87897470cadb824be532e43fe62c17e9acd8987
I tried some PoC for the VM model, but I'm not sure about a few points. What do you think? Should we extract it to a separate module or implicitly use it as in the example

Reply:
@jeremyevans, @fdr What do you think about this approach? https://github.com/ubicloud/ubicloud/commit/b87897470cadb824be532e43fe62c17e9acd8987

Reply:
I like that approach.

Reply:
I refactored it to the new approach. @fdr -300 lines 🔥